From 624eab33866c2fe9e1ec49823a8c9b9b1df8696e Mon Sep 17 00:00:00 2001
From: Santan Kumar <santan.kumar@nxp.com>
Date: Fri, 23 Jun 2017 15:40:27 +0530
Subject: [PATCH 073/211] board/ls2081ardb: Some updates like flash type

Patch makes below changes:
-Update QSPI flash type
	As per updated board design,
	different QSPI flash is connected on boards,
	hence change QSPI flash type from
	Micron n25q512a device to spansion s25fs512s
	device in dts and config.
-Update QIXIS_QMAP_MASK value from 0x07 to 0x70
-Update checkboard()
	display of boot source and board information
-Correct DSPI flash config selection
-Remove deprecated config : CONFIG_DISPLAY_BOARDINFO_LATE

Signed-off-by: Santan Kumar <santan.kumar@nxp.com>
Signed-off-by: Yogesh Gaur <yogeshnarayan.gaur@nxp.com>
Signed-off-by: Priyanka Jain <priyanka.jain@nxp.com>
---
 arch/arm/dts/fsl-ls2081a-rdb.dts        |    4 ++--
 board/freescale/ls2080ardb/ls2080ardb.c |   13 ++-----------
 include/configs/ls2080ardb.h            |    9 ++-------
 3 files changed, 6 insertions(+), 20 deletions(-)

diff --git a/arch/arm/dts/fsl-ls2081a-rdb.dts b/arch/arm/dts/fsl-ls2081a-rdb.dts
index 6489362..aa4aa68 100644
--- a/arch/arm/dts/fsl-ls2081a-rdb.dts
+++ b/arch/arm/dts/fsl-ls2081a-rdb.dts
@@ -41,7 +41,7 @@
 	bus-num = <0>;
 	status = "okay";
 
-	qflash0: n25q512a@0 {
+	qflash0: s25fs512s@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
 		compatible = "spi-flash";
@@ -49,7 +49,7 @@
 		reg = <0>;
 	};
 
-	qflash1: n25q512a@1 {
+	qflash1: s25fs512s@1 {
 		#address-cells = <1>;
 		#size-cells = <1>;
 		compatible = "spi-flash";
diff --git a/board/freescale/ls2080ardb/ls2080ardb.c b/board/freescale/ls2080ardb/ls2080ardb.c
index 376f02d..681d40f 100644
--- a/board/freescale/ls2080ardb/ls2080ardb.c
+++ b/board/freescale/ls2080ardb/ls2080ardb.c
@@ -71,36 +71,27 @@ int checkboard(void)
 #ifdef CONFIG_TARGET_LS2081ARDB
 #ifdef CONFIG_FSL_QIXIS
 	sw = QIXIS_READ(arch);
-	printf("Board Arch: V%d, ", sw >> 4);
 	printf("Board version: %c, ", (sw & 0xf) + 'A');
 
 	sw = QIXIS_READ(brdcfg[0]);
 	sw = (sw & QIXIS_QMAP_MASK) >> QIXIS_QMAP_SHIFT;
 	switch (sw) {
 	case 0:
+	case 4:
 		puts("boot from QSPI DEV#0\n");
-		puts("QSPI_CSA_1 mapped to QSPI DEV#1\n");
 		break;
 	case 1:
 		puts("boot from QSPI DEV#1\n");
-		puts("QSPI_CSA_1 mapped to QSPI DEV#0\n");
 		break;
 	case 2:
-		puts("boot from QSPI EMU\n");
-		puts("QSPI_CSA_1 mapped to QSPI DEV#0\n");
-		break;
 	case 3:
 		puts("boot from QSPI EMU\n");
-		puts("QSPI_CSA_1 mapped to QSPI DEV#1\n");
-		break;
-	case 4:
-		puts("boot from QSPI DEV#0\n");
-		puts("QSPI_CSA_1 mapped to QSPI EMU\n");
 		break;
 	default:
 		printf("invalid setting of SW%u\n", sw);
 		break;
 	}
+	printf("FPGA: v%d.%d\n", QIXIS_READ(scver), QIXIS_READ(tagdata));
 #endif
 	puts("SERDES1 Reference : ");
 	printf("Clock1 = 100MHz ");
diff --git a/include/configs/ls2080ardb.h b/include/configs/ls2080ardb.h
index 7819a87..4cacf6f 100644
--- a/include/configs/ls2080ardb.h
+++ b/include/configs/ls2080ardb.h
@@ -18,7 +18,6 @@
 #define CONFIG_QIXIS_I2C_ACCESS
 #endif
 #define CONFIG_SYS_I2C_EARLY_INIT
-#define CONFIG_DISPLAY_BOARDINFO_LATE
 #endif
 
 #define I2C_MUX_CH_VOL_MONITOR		0xa
@@ -264,7 +263,7 @@ unsigned long get_board_sys_clk(void);
 
 #ifdef CONFIG_TARGET_LS2081ARDB
 #define CONFIG_FSL_QIXIS	/* use common QIXIS code */
-#define QIXIS_QMAP_MASK			0x07
+#define QIXIS_QMAP_MASK			0x70
 #define QIXIS_QMAP_SHIFT		5
 #define QIXIS_LBMAP_DFLTBANK		0x00
 #define QIXIS_LBMAP_QSPI		0x00
@@ -293,19 +292,15 @@ unsigned long get_board_sys_clk(void);
 /* SPI */
 #if defined(CONFIG_FSL_QSPI) || defined(CONFIG_FSL_DSPI)
 #define CONFIG_SPI_FLASH
-#ifdef CONFIG_FSL_QSPI
+#ifdef CONFIG_FSL_DSPI
 #define CONFIG_SPI_FLASH_STMICRO
 #endif
 #ifdef CONFIG_FSL_QSPI
-#ifdef CONFIG_TARGET_LS2081ARDB
-#define CONFIG_SPI_FLASH_STMICRO
-#else
 #define CONFIG_SPI_FLASH_SPANSION
 #endif
 #define FSL_QSPI_FLASH_SIZE		SZ_64M	/* 64MB */
 #define FSL_QSPI_FLASH_NUM		2
 #endif
-#endif
 
 /*
  * RTC configuration
-- 
1.7.1

